src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -328,15 +326,14 @@ public GetResult get(Get get) throws EngineException {
 
     @Override
     public void create(Create create) throws EngineException {
-        final IndexWriter writer;
         try (InternalLock _ = readLock.acquire()) {
-            writer = currentIndexWriter();
+            ensureOpen();
             if (create.origin() == Operation.Origin.RECOVERY) {
                 // Don't throttle recovery operations
-                innerCreate(create, writer);
+                innerCreate(create, indexWriter);
```

Comment:
Can we remove the IW param from innerCreate/Index/Delete now?  They can just use the final instance variable directly?